plugins/providers/bigquery/provider.go

```diff
@@ -120,32 +122,40 @@ func (p *Provider) GetResources(pc *domain.ProviderConfig) ([]*domain.Resource,
 	resourceTypes := pc.GetResourceTypes()
 
 	resources := []*domain.Resource{}
-	ctx := context.Background()
+	eg, ctx := errgroup.WithContext(context.TODO())
+
 	datasets, err := client.GetDatasets(ctx)
 	if err != nil {
 		return nil, err
 	}
 	for _, d := range datasets {
-		dataset := d.ToDomain()
-		dataset.ProviderType = pc.Type
-		dataset.ProviderURN = pc.URN
-
-		if containsString(resourceTypes, ResourceTypeDataset) {
-			resources = append(resources, dataset)
-		}
-
-		if containsString(resourceTypes, ResourceTypeTable) {
-			tables, err := client.GetTables(ctx, dataset.Name)
-			if err != nil {
-				return nil, err
+		d := d
+		eg.Go(func() error {
+			dataset := d.ToDomain()
+			dataset.ProviderType = pc.Type
+			dataset.ProviderURN = pc.URN
+
+			if containsString(resourceTypes, ResourceTypeDataset) {
+				resources = append(resources, dataset)
 			}
-			for _, t := range tables {
-				table := t.ToDomain()
-				table.ProviderType = pc.Type
-				table.ProviderURN = pc.URN
-				dataset.Children = append(dataset.Children, table)
+
+			if containsString(resourceTypes, ResourceTypeTable) {
+				tables, err := client.GetTables(ctx, dataset.Name)
+				if err != nil {
+					return fmt.Errorf("fetching tables for dataset %q: %w", dataset.URN, err)
+				}
+				for _, t := range tables {
+					table := t.ToDomain()
+					table.ProviderType = pc.Type
+					table.ProviderURN = pc.URN
+					dataset.Children = append(dataset.Children, table)
+				}
 			}
```

Comment:
How many goroutines does this spawn? 
Is it possible to limit the number of goroutines spawned by using buffered channels here ?

Reply:
it will be as many datasets there. `errgroup` has `.SetLimit` feature can be use to limit active goroutines within the group, any suggestion for the value we should set?